scripts/test-aws-creds.md

```diff
@@ -0,0 +1,85 @@
+for tests which run longer than `60 minutes` to avoid AWS credentials expiration.
+
+A good followup question for above background refresh will be **"Why didn't
+you increase the ASSUME_ROLE_SESSION_DURATION to maximum allowed 12 hours?
+which should cover most cases"**
+
+The reason ACK team did not rely on increasing `ASSUME_ROLE_SESSION_DURATION`
+is because it is a security risk to have long lived credentials and there is
+a limitation of maximum `3600 seconds` session duration when assuming IAM Role
+using another IAM Role.
+
+Since, ACK prow jobs get default credentials by assuming an IAM role (IRSA),
+credentials acquired by further assuming the test IAM role provided by service
+teams expire within `3600 seconds`. Hence the background refresh is a necessity
+for running service controllers in ACK prow jobs.
+
+For local testing, the background refresh was kept just to have the same
+approach as ACK prow jobs instead of increasing maximum assume role duration
+for `ACK_ROLE_ARN`
+
+### Test Container
+`Test Container` executes the `run-tests.sh` script which runs python e2e tests.
+One difference between `Service Controller` and `Test Container` is that `Test
+Container` cannot be restarted to refresh credentials using a background thread.
+
+To overcome the credential expiry inside `Test Container`, ACK uses the
+`AWS profile` from `~/.aws/credentials` file . When using `AWS profiles`
+AWS SDK automatically refreshes the credentials before their expiry.
+
+To run `Test Container` inside `Prow`, `build-run-test-dockerfile.sh` script mounts
+`~/.aws/credentials` file inside `Test Container` with an `"ack-test"` profile which
+is used to run the e2e tests. The source profile for this `"ack-test"` profile is the
+`IRSA` identity of Prow job pod. See this [template](./templates/prow-test-aws-creds-template.txt)
+for detailed `credentials` file.
+
+When running `Test Container` locally, the local environment will not always have
+consistent `IRSA` source identity as seen in `Prow` environment. To enable credentials
+refresh when running locally, `~/.aws/credentials` file is mounted inside
+`Test Container` with `"ack-test"` AWS profile similar to `Prow` environment but
+the content of local `~/.aws/credentials` file is also copied inside this mounted
+file. See this [template](./templates/local-test-aws-creds-template.txt) for more
+details.
+
+When running `Test Container` locally, the source profile for `"ack-test"` AWS profile
+is provided using `ACK_TEST_SOURCE_AWS_PROFILE` environment variable which defaults
+to `"default"`. 
+
+NOTE
+```
+The source profile for `ack-test` profile is referred from local
+"~/.aws/credentials" file since it's content gets copied into "credentials" file of
+"Test Container".
+```
+
+NOTE
+```
+The AWS profile approach used for "Test Container" is not used inside "Service Controller"
+because currently ACK service controllers do not support mounting AWS "credentials" file.
```

Comment:
Should we adapt the common ACK runtime to support the credentials file?

Reply:
I believe it falls into this bucket(https://github.com/aws-controllers-k8s/community/issues/1041)